src/find/matchers/type_matcher.rs

```diff
@@ -5,70 +5,89 @@
 // https://opensource.org/licenses/MIT.
 
 use std::error::Error;
-use std::fs::FileType;
-use walkdir::DirEntry;
 
-#[cfg(unix)]
-use std::os::unix::fs::FileTypeExt;
-
-use super::{Matcher, MatcherIO};
+use super::{FileType, Follow, Matcher, MatcherIO, WalkEntry};
 
 /// This matcher checks the type of the file.
 pub struct TypeMatcher {
-    file_type_fn: fn(&FileType) -> bool,
+    file_type: FileType,
+}
+
+fn parse(type_string: &str) -> Result<FileType, Box<dyn Error>> {
+    let file_type = match type_string {
+        "f" => FileType::Regular,
+        "d" => FileType::Directory,
+        "l" => FileType::Symlink,
+        "b" => FileType::BlockDevice,
+        "c" => FileType::CharDevice,
+        "p" => FileType::Fifo, // named pipe (FIFO)
+        "s" => FileType::Socket,
+        // D: door (Solaris)
+        "D" => {
+            return Err(From::from(format!(
+                "Type argument {type_string} not supported yet"
+            )))
```

Comment:
Should we open a new issue to track this? [The GNU documentation](https://www.gnu.org/software/findutils/manual/html_node/find_html/Type.html) only describes `-type D` as `door (Solaris)`.

Reply:
Doors are an old IPC mechanism on Solaris.  They get their own file type.

Reply:
I've never used them but there are some old articles describing them, e.g. http://www.kohala.com/start/papers.others/doors.html

I'm not sure whether they're still in use on modern Solaris/Illumos.  I also don't think Rust's stdlib has any support for detecting doors.  I personally wouldn't worry about them.  This code was just moved from here: https://github.com/uutils/findutils/blob/465856ce2454605f7acacb0c8d4189bdbfc1c3d6/src/find/matchers/type_matcher.rs#L33-L37

Reply:
thanks! It is indeed old and difficult to achieve, let it stay that way. :)